tips/TIP-0019/tip-0019.md

```diff
@@ -0,0 +1,2986 @@
+
+### How does it affect other parts of the protocol?
+
+The `dust protection` only affects "value-transactions", since messages with indexation payload, formerly known as "data-transactions", are not stored in the ledger state and are therefore pruned after some time. Milestone payloads are also not affected. All output types like e.g. smart contract requests are affected by the `dust protection`, and must comply with the `min_deposit_of_output` criteria. Therefore these requests could get quite expensive for the user, but the same mechanism introduced for [Microtransactions on Layer 1](#Microtransactions-on-Layer-1) can be utilized for smart contract requests as well.
+
+### Byte cost calculations
+
+To limit the maximum database size, the total IOTA supply needs to be divided by the target database size in bytes to get the worst case scenario regarding the byte costs.
+
+Since this would be way too expensive and would not represent the real distribution of funds on the UTXOs, the zipf distribution fitting of the current ledger state could be utilized. Unfortunately these zipf fittings will not match the future distribution of the funds for several reasons:
+
+- There is already another `dust protection` in place, which distorts the fittings.
+- The fitting was made for mana simulations and took only the richest 100 addresses into account because the zipf distribution is an empirical law that holds only true for the top range of the funds distribution.
+- With new use cases enabled by the new dust protection (e.g. tokenization, storing arbitrary data in the ledger), the distribution will dramatically change.
+- Fittings for other DLT projects do not match because there are transaction fees in place, which decrease the amount of dust outputs in the distribution.
+
+Another possibility would be to estimate how much percentage of the database will be used for outputs with minimum required deposit in the future. Since a fund sparsity percentage of more than 20% would already be bad for other upcoming protocol features like the mana calculation, we could take this value for our calculation instead of the worst case.
+
+### Weights for different outputs
+
+The different output types mentioned in the [Output Types Draft TIP-18](https://github.com/iotaledger/tips/pull/38) contain several mandatory and optional fields. Every field itself creates individual computational and storage requirements for the node, which is considered by having different weights for every field.
+
+##### Field types
+
+The following table describes different field types in an output:
+
+<table>
+    <tr>
+        <th>Name</th>
+        <th>Description</th>
+        <th>Weight</th>
```

Comment:
Maybe we should specify that these can be floating point / fractional values. `vbyte_cost` on the other hand is of type integer, right? Maybe we should specify that somewhere too?